documentation/error-handling.md

```diff
@@ -0,0 +1,113 @@
+# Error handling
+
+Because UI extension scripts run in non-browser environments (Web Worker in web, JavascriptCore on Mobile), it is recommended to manually set up error listeners and error reporting. However, it is possible to use external libraries for error reporting.
+
+## Error handling in UI extensions
+
+UI extensions can throw two types of error: lifecycle errors and non-lifecycle errors.
+
+### Lifecycle error
+
+Most errors will fall into this type. These errors can be thrown when:
+
+- a UI extension script is loaded or imported
+- the UI extension script is run
+- during any callback interaction with UI extension components or APIs
+
+To catch these errors, add an event listener for `unhandledrejection`.
+
+  ```javascript
+  self.onunhandledrejection = (error) => {
+    console.warn('onunhandledrejection', error);
+  };
+  // or
+  self.addEventListener('unhandledrejection', (error) => {
+    console.warn('event unhandledrejection', error);
+  });
+  ```
+
+### Non-lifecycle error
```

Comment:
I don't think we should break these down this way. They do technically happen at these times, but that is just a factor of our implementation. For a developer, what they need to know is just that they can add the same global event listeners for untaught exceptions (the error event) and for unhandled promise rejections, just like they can in other JS environments.